src/config/mod.rs

```diff
@@ -21,29 +21,28 @@ pub const JSON_RPC_VERSION: &str = "2.0";
 /// The top-level struct representing the config. Calls to [`Config::from_file`] deserialize into
 /// this struct.
 #[derive(Deserialize)]
-pub(crate) struct Config {
+pub struct Config {
     pub server: Server,
     pub subnets: HashMap<String, Subnet>,
 }
 
 impl Config {
     /// Reads a TOML configuration in the `s` string and returns a [`Config`] struct.
-    pub fn from_str(s: &str) -> Result<Self> {
-        let config = toml::from_str(&s)?;
+    pub fn from_string(s: &str) -> Result<Self> {
```

Comment:
Just curious, why this rename? If it's `from_str`, then why not implement [FromStr](https://doc.rust-lang.org/std/str/trait.FromStr.html), and if that doesn't feel right, then why not call it `from_toml` or `from_toml_str`? I suppose mentioning `toml` in the method name would also put `from_file` into question.